### PR TITLE
MySQL: Add option to allow cleartext passwords

### PR DESCRIPTION
**Pull request by enginecan**
Adds "Allow Cleartext Passwords" checkbox to MySQL connection settings page in order to enable support for Cleartext Passwords

image

Rework of #30112  Fixes https://github.com/grafana/grafana/discussions/37010

### CODE REVIEW

**Comment by gannaramu** [comment]
also addresses https://github.com/grafana/grafana/discussions/37010

**Comment by zoltanbedi** [comment]
Hey @enginecan,
Thanks for creating this PR. Are you interested in wrapping this up? Maybe I can help you with something?

**Comment by enginecan** [comment]
Hey @zoltanbedi, I've solved the merge conflicts. Feel free to review...

**Comment by zoltanbedi** [comment]
Thanks @enginecan,
Looks like the CI is failing on lint issues please resolve those and this new option needs to be added to the documentation as well. Can you please do those?

**Comment by enginecan** [comment]
@zoltanbedi - Could you please fix those issues? I don't have enough time to spend on linting issues.

**Comment by zoltanbedi** [comment]
@enginecan yep I can continue.

File: docs/sources/datasources/mysql/_index.md [hunk 1]
```diff
@@ -35,19 +35,20 @@ Administrators can also [configure the data source via YAML]({{< relref "#provis
...
+| **Max lifetime**              | The maximum amount of time in seconds a connection may be reused, default `14400`/4 hours. This should always be lower than configured [wait_timeout](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_wait_timeout) in MySQL (Grafana v5.4+).                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (mdvictor) [inline docs/sources/datasources/mysql/_index.md:51]: Nit: space things here for better visibility

Comment (zoltanbedi) [reply docs/sources/datasources/mysql/_index.md]: I've added a line break but I felt like that it isn't that bad when seen on the rendered page.

**Review by lwandz13 — APPROVED** [review]
LGTM

**Review by mdvictor — APPROVED** [review]
lgtm

**Review by baldm0mma — APPROVED** [review]
LGTM!

File: public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx [hunk 2]
```diff
@@ -150,6 +150,26 @@ export const ConfigurationEditor = (props: DataSourcePluginOptionsEditorProps<My
             value={jsonData.tlsSkipVerify || false}
           ></InlineSwitch>
         </InlineField>
+        <InlineField
+          labelWidth={WIDTH_MEDIUM}
+          tooltip={
+            <span>
+              Allows using the cleartext client side plugin if required by an account, such as one defined with the PAM
```
Comment (baldm0mma) [inline public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx:157]: I believe cleartext passwords were introduced in mysql 5.5.12; is it worth noting in the tooltip that it won't work for versions earlier than that?

Comment (zoltanbedi) [reply public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx:157]: I think that whoever is configuring and enabling this is aware of that.

**Comment by zoltanbedi** [comment]
Thanks for your contribution @enginecan!

**Comment by enginecan** [comment]
> Thanks for your contribution @enginecan!

Likewise, thanks for creating this great tool!

**Pull request merged**